src/dev/build/tasks/copy_source_task.ts

```diff
@@ -6,65 +6,68 @@
-        'src/**',
-
-        'x-pack/plugins/**',
-        'x-pack/.i18nrc.json',
-        'x-pack/package.json',
+    const select = [
+      'yarn.lock',
+      '.npmrc',
+      'config/kibana.yml',
+      'config/node.options',
+      '.i18nrc.json',
+      'src/**',
 
-        '!{src,x-pack}/**/*.{test,test.mocks,mock,mocks}.*',
-        '!{src,x-pack}/**/target/**',
-        '!{src,x-pack}/**/{__stories__,storybook,.storybook}/**',
-        '!{src,x-pack}/**/{test,tests,test_resources,test_data,__tests__,manual_tests,__jest__,__snapshots__,__mocks__,mock_responses,mocks,fixtures,__fixtures__,cypress,integration_tests}/**',
+      'x-pack/plugins/**',
+      'x-pack/.i18nrc.json',
+      'x-pack/package.json',
 
-        '!src/dev/**',
-        '!src/**/mocks.{js,ts}',
-        '!src/cli/repl/**',
-        '!src/cli*/dev.js',
-        '!src/plugins/telemetry/schema/**',
-        '!src/core/server/core_app/assets/favicons/favicon.distribution.{ico,png,svg}',
-        '!src/functional_test_runner/**',
-        '!src/setup_node_env/index.js',
+      '!src/dev/**',
+      '!src/**/mocks.{js,ts}',
+      '!src/cli*/dev.js',
+      '!src/plugins/telemetry/schema/**',
+      '!src/core/server/core_app/assets/favicons/favicon.distribution.{ico,png,svg}',
+      '!src/setup_node_env/index.js',
 
-        '!x-pack/plugins/**/{ftr_e2e,e2e}/**',
-        '!x-pack/plugins/**/scripts/**',
-        '!x-pack/plugins/telemetry_collection_xpack/schema/**',
+      '!x-pack/plugins/telemetry_collection_xpack/schema/**',
 
-        '!**/jest.config.js',
-        '!**/jest.config.dev.js',
-        '!**/jest.integration.config.js',
-        '!**/jest_setup.{js,ts}',
-        '!**/*.{story,stories}.{js,ts}',
-        '!**/test_mocks.ts',
-        '!**/*.{sh,md,mdx,asciidoc}',
-        '!**/*.console',
-        '!**/*.scss',
-        '!**/*.docnav.json',
-        '!**/{dev_docs,docs}/**',
-        '!**/public/**/*.{js,ts,tsx,json}',
+      '!**/jest*',
+      '!**/*.{story,stories}.{js,ts}',
+      '!**/test_mocks.ts',
+      '!**/*.{scss,console,d.ts,sh,md,mdx,asciidoc,docnav.json}',
+      '!**/*.{test,test.mocks,mock,mocks,spec}.*',
+      '!**/{packages,dev_docs,docs,public,__stories__,storybook,.storybook,ftr_e2e,e2e,scripts,test,tests,test_resources,test_data,__tests__,manual_tests,__jest__,__snapshots__,__mocks__,mock_responses,mocks,fixtures,__fixtures__,cypress,integration_tests}/**',
+    ];
 
-        // explicitly ignore all bazel package locations, even if they're not selected by previous patterns
-        ...(await discoverBazelPackages()).map((pkg) => `!${pkg.normalizedRepoRelativeDir}/**`),
-      ],
+    const piscina = new Piscina({
+      filename: resolve(__dirname, 'copy_source_worker.js'),
     });
+
+    const globbyOptions = { cwd: config.resolveFromRepo('.') };
+    (
+      await Promise.all([
+        globby(select, globbyOptions),
+        globby(
+          [
+            '{x-pack,src}/plugins/*/public/assets/**',
```

Comment:
This second step is mainly in support of excluding all public folders, except `/public/assets`.  ~7mb of  media files get pulled in through webpack that get duplicated otherwise.